src/client/pkg/grpcutil/dialer.go

```diff
@@ -35,13 +35,17 @@ func newDialer(opts ...grpc.DialOption) *dialer {
 func (d *dialer) Dial(addr string) (*grpc.ClientConn, error) {
 	d.lock.Lock()
 	defer d.lock.Unlock()
+
 	if conn, ok := d.connMap[addr]; ok {
 		return conn, nil
 	}
 	opts := append(d.opts,
 		grpc.WithUnaryInterceptor(tracing.UnaryClientInterceptor()),
 		grpc.WithStreamInterceptor(tracing.StreamClientInterceptor()),
 	)
+	if (strings.Index(addr, "dns:///") == -1) {
```

Comment:
It'd be slightly faster to [just check for the prefix](https://golang.org/pkg/strings/#HasPrefix). If `dns:///` is in the string and it's not the prefix, it's going to be broke anyways.

Reply:
Yeah. The updated code has changed it to HasPrefix